### DISCUSSION TITLE
B站链接拖入播放的视频总是480p

### DISCUSSION THREAD

**Discussion by ghost** [hooke007/mpv_PlayKit#231] [mpv 通用]

**Comment by ghost** [comment]
解决了，yt-dlp.conf加上--cookies-from-browser edge,与yt放在同一目录

**Reply by hooke007** [reply]
mpv有选项支持直接传递
https://github.com/hooke007/MPV_lazy/blob/1f86aea9424bdd3bebe9436d5ebac88a54b843ec/portable_config/mpv.conf#L210-L212

**Reply by ghost** [reply]
大大，这个是要下载cookies吗？还是写成ytdl-raw-options-append =cookies-from-browser edge，我不太懂

**Reply by hooke007** [reply]
```ini
ytdl-raw-options-append = cookies-from-browser=edge
```

**Reply by ghost** [reply]
好的，谢谢大大

**Reply by OmiChrome** [reply]
> ```ini
> ytdl-raw-options-append = cookies-from-browser=edge
> ```

how can i make it for firefox? or opera GX? 
 i don't understand Chinese  i read the discussion through translate :)

**Reply by M1saka10010** [reply]
> > ```ini
> > ytdl-raw-options-append = cookies-from-browser=edge
> > ```
> 
> how can i make it for firefox? or opera GX? i don't understand Chinese i read the discussion through translate :)

It seems can't work with other browser. At least it can't work with chrome. You can login ur account in edge then use it. (Most websites can log in from multiple places at the same time). Hope u can understand my words, I am not native english speaker.

**Comment by chenxi492** [comment]
想问一下，我和楼主问题一致，但是这样做了之后控制台显示
[ytdl_hook] ERROR: 'latin-1' codec can't encode characters in position 548-549: ordinal not in range(256)
[ytdl_hook] youtube-dl failed: unexpected error occurred
Failed to recognize file format.
所以后续该怎么做呢ww，麻烦了

**Reply by ghost** [reply]
你是不是改了ytdl_hookz.conf,要改播放的清晰度，在mpv.conf加入I（图一）。如果是利用yt下载视频，在yt的同目录下创建yt-dlp.conf(图二）
![image](https://user-images.githubusercontent.com/89686335/208607950-55238630-3bc3-4e9e-92e8-14c334e88014.png)
![image](https://user-images.githubusercontent.com/89686335/208608172-ea9b02a1-5e00-49d7-a59d-67bdd98cd7ac.png)

**Reply by chenxi492** [reply]
我的写法和你的是一致的，就是添加过ytdl-raw-options-append = cookies-from-browser=edge后再去拉链接，才出现我提到的报错，不添加这行语法mpv能正常打开视频，但是清晰度只有480p

**Reply by ghost** [reply]
> 我的写法和你的是一致的，就是添加过ytdl-raw-options-append = cookies-from-browser=edge后再去拉链接，才出现我提到的报错，不添加这行语法mpv能正常打开视频，但是清晰度只有480p

你edge里登了账号没？我也不清楚，小白一个

**Reply by chenxi492** [reply]
登录了，也不知道问题在哪麻了，我也小白
等等看有没有大佬回复了，感谢！

**Reply by Arxhexa** [reply]
我下面写了一个可能的解决方法，不用走ytdl的视频解析

**Comment by Arxhexa** [comment]
之前看到过 [这个](https://github.com/diannaojiang/Bilibili-Playin-Mpv) 东西，配合 [Bilibili-Evolved](https://github.com/the1812/Bilibili-Evolved) 使用
不过我没试到底能不能用，理论上应该是可以的，参数给到mpv的是直链
但是上面这个应该是没弹幕的，我自己是用 [bilibili-api](https://github.com/Nemo2011/bilibili-api) 分别拉视频流音频流和弹幕文件最后扔进mpv

**Reply by chenxi492** [reply]
感谢告知，我在那之后也找了一个[脚本](https://github.com/LuckyPuppy514/Play-With-MPV)，效果可能和你的差不多，也有配套的[弹幕脚本](https://github.com/itKelis/MPV-Play-BiliBili-Comments)，问题也算是变相解决了吧，总之感谢帮助

**Comment by M1saka10010** [comment]
~~很神奇，我edge上并没有登录B站号，但是居然能用~~我填chrome反而不行（刚开edge发现是登了的
